.NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/BaseCurrencyParser.cs

```diff
@@ -73,19 +73,50 @@ private static bool CheckUnitsStringContains(string fractionUnitCode, string fra
 
         private string GetResolutionStr(object value)
         {
+            // Nothing to resolve. This happens when the entity is a currency name only (no numerical value).
+            if (value == null)
+            {
+                return null;
+            }
+
             return Config.CultureInfo != null ?
                 ((double)value).ToString(Config.CultureInfo) :
                 value.ToString();
         }
 
+        private void AddToResult(ParseResult result, string mainUnitIsoCode, object numberValue,
```

Comment:
Can you move results.Add(result); out of the method? The method could be called something like CreateCurrencyResult and return a result.
It is a bad design to have methods that internally change the state of one passed parameter. Especially if the parameter is not really necessary to the method body.  If a parameter is really necessary, it should at least be explicitly marked as being used by reference.